### PR TITLE
Added guard checks before typecast

### PR DESCRIPTION
**Pull request by amandeeprehal-lab49**
Added guard checks before typecast

### CODE REVIEW

File: contracts/common/BaseHTS.sol [hunk 1]
```diff
@@ -37,43 +16,51 @@ contract BaseHTS is HederaTokenService, IBaseHTS {
...
-    function associateTokensPublic(
-        address account,
-        address[] memory tokens
-    ) external override returns (int256 responseCode) {
-        return HederaTokenService.associateTokens(account, tokens);
-    }
-
     function mintTokenPublic(
         address token,
         int256 amount
```
Comment (shivamgarglab49) [inline contracts/common/BaseHTS.sol:21]: int256 in mintTokenPublic as argument ?

File: contracts/common/BaseHTS.sol [hunk 2]
```diff
@@ -37,43 +16,51 @@ contract BaseHTS is HederaTokenService, IBaseHTS {
...
             revert("Mint Failed");
         }
 
-        return (responseCodeNew, int256(int256(newTotalSupplyNew)));
+        return (responseCodeNew, newTotalSupplyNew);
     }
 
     function burnTokenPublic(
         address token,
         int256 amount
```
Comment (shivamgarglab49) [inline contracts/common/BaseHTS.sol:45]: same here.

File: deployment/state/contracts.json [hunk 3]
```diff
@@ -212,5 +212,68 @@
...
+    "transparentProxyId": "0.0.13392528"
+  },
+  {
+    "name": "nftholder",
+    "id": "0.0.13392533",
+    "address": "0x0000000000000000000000000000000000cc5a95",
+    "timestamp": "2023-05-24T10:20:20.265Z",
+    "hash": "4d7e43c0d7702071bbf7bea25229e2cb",
+    "transparentProxyAddress": "0x0000000000000000000000000000000000cc5a97",
+    "transparentProxyId": "0.0.13392535"
```
Comment (shivamgarglab49) [inline deployment/state/contracts.json:277]: please remove older entries, also update the UAT json.

File: contracts/common/BaseHTS.sol (hunk 2 above)
Comment (amandeeprehal-lab49) [reply contracts/common/BaseHTS.sol]: fixed it

File: deployment/state/contracts.json (hunk 3 above)
Comment (amandeeprehal-lab49) [reply deployment/state/contracts.json]: Done

File: contracts/common/BaseHTS.sol [hunk 4]
```diff
@@ -37,43 +16,51 @@ contract BaseHTS is HederaTokenService, IBaseHTS {
...
     function mintTokenPublic(
         address token,
-        int256 amount
-    ) external override returns (int256 responseCode, int256 newTotalSupply) {
+        uint256 amount
+    ) external override returns (int256 responseCode, int64 newTotalSupply) {
         bytes[] memory metadata;
 
+        require(
+            amount >= 0 && amount <= INT_64_MAX_VALUE,
```
Comment (shivamgarglab49) [inline contracts/common/BaseHTS.sol:26]: amount >= 0, it should be > only .. ?

File: contracts/common/BaseHTS.sol [hunk 5]
```diff
@@ -37,43 +16,51 @@ contract BaseHTS is HederaTokenService, IBaseHTS {
...
-        int256 amount
-    ) external override returns (int256 responseCode, int256 newTotalSupply) {
+        uint256 amount
+    ) external override returns (int256 responseCode, int64 newTotalSupply) {
         int64[] memory serialNumbers;
-        (int256 responseCodeNew, int64 newTotalSupplyNew) = HederaTokenService
-            .burnToken(token, int64(amount), serialNumbers);
+
+        require(
+            amount >= 0 && amount <= INT_64_MAX_VALUE,
```
Comment (shivamgarglab49) [inline contracts/common/BaseHTS.sol:50]: same here.

**Review by shivamgarglab49 — APPROVED** [review]

File: contracts/common/BaseHTS.sol (hunk 4 above)
Comment (amandeeprehal-lab49) [reply contracts/common/BaseHTS.sol]: Done

File: contracts/common/BaseHTS.sol (hunk 5 above)
Comment (amandeeprehal-lab49) [reply contracts/common/BaseHTS.sol]: Done

**Pull request merged**